.github/workflows/prepare-new-pr.yml

```
@@ -0,0 +1,21 @@
+name: 'Prepare new PR'
+on:
+  pull_request_target:
+    types: [opened, reopened]
+    branches: [ 'main*' ]
+
+jobs:
+  prepare-new-pr:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      pull-requests: write
+    if: ${{ github.repository_owner == 'open-telemetry' }}
+    steps:
+      - uses: actions/checkout@v4
```

I think this defaults to checking out `main` after switching to `pull_request_target` (I suspect due to the security concerns), I think need:

```suggestion
      - uses: actions/checkout@v4
        with:
          ref: ${{ github.head_ref }}
```